cinema/views.py

```diff
@@ -56,3 +76,58 @@
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        movie_id = self.request.query_params.get("movie")
+        date = self.request.query_params.get("date")
+        if movie_id:
+            movie_id = int(movie_id)
+            queryset = queryset.filter(movie_id=movie_id)
+        if date:
+            date = date.split("-")
+            queryset = queryset.filter(
+                show_time__year=int(date[0]),
+                show_time__month=int(date[1]),
+                show_time__day=int(date[2]),
+            )
+        if self.action == "list":
+            queryset = queryset.annotate(
+                tickets_available=F("cinema_hall__rows")
+                * F("cinema_hall__seats_in_row")
+                - Count("tickets")
```

Comment:
The calculation for 'tickets_available' in the 'MovieSessionViewSet' assumes that all rows and seats are available. Ensure that this logic aligns with the actual business rules for ticket availability, as it might not account for reserved or blocked seats.